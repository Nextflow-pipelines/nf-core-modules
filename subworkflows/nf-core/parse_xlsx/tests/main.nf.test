// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core subworkflows test prep_reference
nextflow_workflow {

    name "Test Subworkflow PARSE_XLSX"
    script "../main.nf"
    workflow "PARSE_XLSX"

    // TODO nf-core: Add tags for all modules used within this subworkflow. Example:
    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/parse_xlsx"
    tag "subworkflows/parse_xlsx"
    tag "parsexlsx/concs"
    tag "parsexlsx/samples"
    tag "parsexlsx/customer"
    tag "parsexlsx/organism"
    tag "parsexlsx/experiments"

    test("cage - pe - xlsx") {

        when {
            workflow {
                """
                // TODO nf-core: define inputs of the workflow here. Example:
                input[0] = file(params.test_data['generic']['xlsx']['test_xlsx'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
                //TODO nf-core: Add all required assertions to verify the test output.
                { assert workflow.out.samples ==~ "samplesheet.csv" },
                { assert workflow.out.concs ==~ "control_case.csv" },
                { assert workflow.out.informations ==~ "*.yml" },
                { assert snapshot(workflow.out.versions).match("parse_xlsx") }
            )
        }
    }
}
