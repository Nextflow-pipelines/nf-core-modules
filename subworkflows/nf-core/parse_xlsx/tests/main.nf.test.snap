{
    "cage - pe - xlsx": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "concs": [
                    
                ],
                "informations": [
                    
                ],
                "samples": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-11T22:26:00.102004"
    },
    "parse_xlsx": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-11T22:26:00.119353"
    }
}